EditorInterface.cpp

```diff
@@ -2986,6 +3095,154 @@ void EditorInterface::IGSquadEditor()
+					static int ChoreoKDeleteNum = -1;
+					static bool AdjustChoreo = false;
+					static bool AdjustPrevChoreoSlots = false;
+					ImGui::PushStyleColor(ImGuiCol_Button, (ImVec4)ImColor::HSV(0.0f, 1.0f, 0.8f));
+					ImGui::PushStyleColor(ImGuiCol_ButtonHovered, (ImVec4)ImColor::HSV(0.0f, 1.0f, 0.9f));
+					ImGui::PushStyleColor(ImGuiCol_ButtonActive, (ImVec4)ImColor::HSV(0.0f, 1.0f, 1.0f));
+					if (ImGui::Button("Delete Tools")) {
+						ImGui::OpenPopup("Delete Options");
+					}
+					ImGui::PopStyleColor(3);
+					if (ImGui::BeginPopup("ChoreoKey flags")) {
+
+						ImGui::Checkbox("Don't rotate around asterix", &ckflags[0]);
+						ImGui::Checkbox("Bit 2", &ckflags[1]);
+						ImGui::Checkbox("Bit 3", &ckflags[2]);
+						ImGui::Checkbox("Formations always have spears out", &ckflags[3]);
+						ImGui::Checkbox("Look at asterix", &ckflags[4]);
+						ImGui::Checkbox("Bit 6", &ckflags[5]);
+						ImGui::Checkbox("Enemies can run", &ckflags[6]);
+						ImGui::Checkbox("Bit 8", &ckflags[7]);
+
+						if (ImGui::Button("OK")) {
+							for (int8_t index = 0; index < 8; index++) {
+								ckey->flags = toggle_bit(ckey->flags, index, ckflags[index]);
+							}
+						}
+						ImGui::EndPopup();
+					}
+
+					if (ImGui::BeginPopup("Delete Options")) {
+
+						ImGui::InputInt("Delete num ChoreoGraphy:", &ChoreoDeleteNum);
+						ImGui::Checkbox("Merge Choreokeys to previous ChoreoGraphy", &AdjustPrevChoreoSlots);
+						//ImGui::Checkbox("Adjust Number of Choreokeys", &AdjustChoreo);
+						ImGui::InputInt("Delete num ChoreoKey:", &ChoreoKDeleteNum);
+						ImGui::Checkbox("Adjust Number of Choreokeys", &AdjustChoreo);
+
+						if (ImGui::Button("OK")) {
+							if (selectedSquad) {
+								IsDoingHomeWork = true; // Extremely Important! PogChamp
+
+								if (ChoreoKDeleteNum > -1) {
+									//ChoreoKey
+									auto* Keytoremove = selectedSquad->choreoKeys[ChoreoKDeleteNum].get();
+									auto ChoreoKDeletePos = selectedSquad->choreoKeys.begin() + ChoreoKDeleteNum;
+									selectedSquad->choreoKeys.erase(ChoreoKDeletePos);
+									kenv.removeObject(Keytoremove);
+									if (AdjustChoreo == true) {
+										selectedSquad->choreographies[getChoreo(ChoreoKDeleteNum)]->numKeys = selectedSquad->choreographies[getChoreo(ChoreoKDeleteNum)]->numKeys - 1;
+									}
+								}
+									
+								if (ChoreoDeleteNum > -1) {
+									// ChoreoGraphy
+									auto* Choreographytoremove = selectedSquad->choreographies[ChoreoDeleteNum].get();
+									auto ChoreoDeletePos = selectedSquad->choreographies.begin() + ChoreoDeleteNum;
+									selectedSquad->choreographies.erase(ChoreoDeletePos);
+									kenv.removeObject(Choreographytoremove);
+									if (AdjustPrevChoreoSlots == true) {
+										selectedSquad->choreographies[getChoreo(ChoreoDeleteNum - 1)]->numKeys = selectedSquad->choreographies[getChoreo(ChoreoDeleteNum - 1)]->numKeys + selectedSquad->choreographies[getChoreo(ChoreoDeleteNum)]->numKeys;
+									}
```

Comment:
So deleting a choreography will move all its keys to the previous choreography? Interesting... 🤔 
Problem: you're using getChoreo with a choreo index but not a key index...
and obviously I can't remove the first choreo, I would have to remove the second instead...